packages/engine/lib/error/src/error.rs

```diff
@@ -0,0 +1,60 @@
+use std::{error::Error, panic::Location};
+
+use crate::{Context, Frame, Message, Report};
+
+#[cfg(feature = "std")]
+impl<E> From<E> for Report<E>
+where
+    E: Error + Send + Sync + 'static,
+{
+    #[track_caller]
+    fn from(error: E) -> Self {
+        #[cfg(nightly)]
+        let backtrace = if error.backtrace().is_some() {
+            None
+        } else {
+            Some(std::backtrace::Backtrace::capture())
+        };
+
+        Self::from_frame(
+            Frame::from_std(error, Location::caller(), None),
+            #[cfg(all(nightly, feature = "std"))]
+            backtrace,
+            #[cfg(feature = "spantrace")]
+            Some(tracing_error::SpanTrace::capture()),
+        )
+    }
+}
+
+/// Extension trait for [`Error`] to provide context information and convert it to a [`Report`].
+#[cfg(feature = "std")]
+pub trait ErrorExt: Error + Sized {
+    /// Converts the [`Error`] to a [`Report<Error>`][Report] and adds a contextual message.
+    fn add_message<M>(self, message: M) -> Report<Self>
```

Comment:
I think we should consider making people convert Errors explicitly into Reports _before_ adding messages. I don't feel strongly though, so if you think this greatly improves ergonomics we can of course keep it.

It feels a bit strange to me that the method you use on reports to add messages, is also the same method you use on errors to convert them into reports and add messages.

Reply:
As discussed off-GitHub, I removed the extension trait for now. If we feel that this convenience is needed, it will be added later